src/game_screen.h

```diff
@@ -247,4 +250,19 @@ inline bool Game_Screen::IsBattleAnimationWaiting() {
 	return (bool)animation;
 }
 
+inline Game_Picture& Game_Screen::GetPicture(int id) {
+	assert(id >= 0);
+
+	PreallocatePictureData(id);
+
+	return pictures[id - 1];
+}
+
+inline void Game_Screen::PreallocatePictureData(int id) {
```

Comment:
Why is there now DoPreallocatePicData and PreallocatePicData?

Reply:
To inline the check for whether or not you should allocate. Since it's expected to almost never happen you can avoid the cost of a function call in the common case.